llvm/include/llvm/IR/RepoDefinition.h

```diff
@@ -107,56 +103,116 @@ struct GODigestState {
   }
-  // Update G's dependencies.
-  GOInfo &GInfo = GOI[G];
-  GInfo.InitialDigest = std::move(Result.InitialDigest);
-  GInfo.Dependencies = std::move(Result.Dependencies);
-}
-
-// Create a global object information map and calculate the number of hashed
-// functions and variable  inside of the Module M.
-/// \param M Called module.
-/// \returns a pair containing a global object information map and a structure
-/// holding the number of hashed global variables and functions.
-std::pair<GOInfoMap, GONumber> calculateGONumAndGOIMap(Module &M);
+
+  /// Construct the global object information map (GOIMap) and calculate the
+  /// number of hashed functions and variables inside of the Module M.
+  void calculateGONumAndGOIMap(const Module &M);
+
+  /// Compute the hash value for the given global object GO.
+  /// \param GO The global object.
+  /// \param HasRepoDefinitionMD true if using the digest of GO's dependencies
+  ///        which is stored in the RepoDefinition metadata.
+  /// \return The global object's digest value.
+  DigestType calculateDigest(const GlobalObject *GO, bool HasRepoDefinitionMD);
+
+  /// Calculate the initial hash value, dependencies and contributions for the
+  /// global object 'G' and store these informaion in the GOImap.
+  /// \param G Calculated global object.
+  template <typename GlobalType> void calculateGOInfo(const GlobalType *G);
+  void calculateGOInfo(const GlobalObject *GO);
+
+  /// Accumulate the GO's digest and get its GODigestState.
+  /// \param GO Calculated global object.
+  /// \param GOHash A object that is used to calculate the GO's hash value.
+  /// \param HasRepoDefinitionMD true if using the digest of GO's dependencies
+  ///        which is stored in the RepoDefinition metadata.
+  /// \returns a structure containing the GO's state.
+  GODigestState accumulateGODigest(const GlobalObject *GO, MD5 &GOHash,
+                                   bool HasRepoDefinitionMD);
+
+  /// Loop through the contributions and add the initial digest of each global
+  /// object inside of the contributions to GOHash.
+  void updateDigestUseContributions(MD5 &GOHash, const GOVec &Contributions);
+
+  /// Loop through the dependencies and add the final digest of each global
+  /// object inside of the dependencies to GOHash.
+  std::tuple<size_t, DigestType>
+  updateDigestUseDependencies(const GlobalObject *GO, MD5 &GOHash,
+                              unsigned GODepth, const GOVec &Dependencies,
+                              bool HasRepoDefinitionMD);
+
+  /// Update the digest of an invidual GO incorporating the hashes of all its
+  /// dependencies and contributions.
+  /// \param GO The global object whose digest is to be computed.
+  /// \param HasRepoDefinitionMD true if using the digest of GO's dependencies
+  ///        which is stored in the RepoDefinition metadata.
+  /// \returns a tuple containing the state information which includes
```

Comment:
I think you could remove the words "the state information which includes" without losing any information.

Reply:
Removed, thanks.